File: 4- Array ADT/linearSearch.cpp
/* Linear Search */
#include <iostream>

struct Array{

	int A[10];
	int size;
	int length;

};

void Display(struct Array arr)
{
	for (int i = 0; i < arr.length; i++)	// Printing elements
		std::cout << "\nA[" << i << "] = " << arr.A[i] << std::endl;
}


void swap(int* x, int* y){
	int temp = *x;
	*x = *y;
	*y = temp;
}


int LinearSearch(struct Array* arr, int key){	//For transposition

	for (int i = 0; i < arr->length; i++){
		if (key == arr->A[i]){
			swap(&arr->A[i], &arr->A[i-1]);
			return i;
		}
	}
	return -1;
}

int main(){

	struct Array arr1 = { {2, 23, 14, 5, 6, 9, 8, 12}, 10, 8 };
	std::cout << LinearSearch(&arr1, 23);
	Display(arr1);
	return 0;
}

File: JavaScript - C#/3- Searching Algorithms/linearSearch.js
// Best -> O(1)
// Average -> O(n)
// Worst -> O(n)

function linearSearch(arr, val) {
  for (let i = 0; i < arr.length; i++) {
    if (arr[i] === val) return i;
  }

  return -1;
}

let arr = [10, 20, 30, 40, 50, 60, 70, 80, 90, 100];

console.log(linearSearch(arr, 20));
console.log(linearSearch(arr, 100));
console.log(linearSearch(arr, 10));
console.log(linearSearch(arr, 31));
